yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -819,12 +819,22 @@ private[spark] class Client(
       javaOpts += "-XX:CMSIncrementalDutyCycle=10"
     }
 
-    // Include driver-specific java options if we are launching a driver
-    if (isClusterMode) {
```

Comment:
why did we move the if (isClusterMode) check?    it seems we are now doing extra checks we don't need to when not in cluster mode or am I missing something